blockchain/scriptval.go

```diff
@@ -107,9 +107,9 @@ out:
 				break out
 			}
 
-			atomic.AddUint32(txVI.txSigChecks, uint32(vm.SigChecks()))
+			txSigChecks := atomic.AddUint32(txVI.txSigChecks, uint32(vm.SigChecks()))
 
-			if v.flags.HasFlag(txscript.ScriptReportSigChecks) && atomic.LoadUint32(txVI.txSigChecks) > MaxTransactionSigChecks {
+			if v.flags.HasFlag(txscript.ScriptReportSigChecks) && txSigChecks > MaxTransactionSigChecks {
```

Comment:
Changed to use return value of the AddUinit32 call above, as this could actually lead to an error being raised with the wrong transaction being reported. =\